### PR TITLE
Add 100% test coverage for Core::ElasticCoreActions

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
Another PR that adds a spec for our core modules - this time for `Core::ElasticCoreActions`.

### CODE REVIEW

File: spec/core/elastic_connector_actions_spec.rb [hunk 1]
```diff
@@ -0,0 +1,558 @@
+require 'core/elastic_connector_actions'
+require 'connectors/connector_status'
+require 'connectors/sync_status'
+require 'utility/es_client'
+
+describe Core::ElasticConnectorActions do
+  let(:connector_id) { 'one-two-three' }
+  let(:es_client) { double }
```
Comment (seanstory) [inline spec/core/elastic_connector_actions_spec.rb:8]: Not for this PR, but has there been any discussion about getting a local Elasticsearch available on the CI instances for connectors? That would make these more "integration" tests than unit tests, but also might significantly lower the mocking burden.

**Review by seanstory — APPROVED** [review]
My eyes glazed over after a few hundred lines, but kudos for writing so many tests in one go!

File: lib/core/elastic_connector_actions.rb [hunk 2]
```diff
@@ -7,10 +7,12 @@
...
 #
 require 'active_support/core_ext/hash'
-require 'utility/elasticsearch/index/mappings'
-require 'utility/elasticsearch/index/text_analysis_settings'
 require 'connectors/connector_status'
 require 'connectors/sync_status'
+require 'utility/elasticsearch/index/mappings'
+require 'utility/elasticsearch/index/text_analysis_settings'
+require 'utility/es_client'
+require 'utility/logger'
```
Comment (wangch079) [inline lib/core/elastic_connector_actions.rb:15]: Why not
```suggestion
require 'utility'
```

I also observed such cases before, even though there's no `require` for some files, it still doesn't throw any error.

**Review by wangch079 — APPROVED** [review]
Nice work

File: spec/core/elastic_connector_actions_spec.rb (hunk 1 above)
Comment (artem-shelkovnikov) [reply spec/core/elastic_connector_actions_spec.rb:8]: We do have integration test that is ran nightly - you can run it with `make ftest` - it's a functional test though, not an integration one.

We did not have a discussion about that and it's something nice to talk about indeed.

IMO there's not much mocking burden as long as the class doesn't do too much - and mocking helps see class dependencies on 3rd-party components when looking at the test, which I think is quite valuable too. 

All in all, I think the strategy I personally like the most is - as many as possible unit tests + a small number of integration tests that test big chunks of the system + a small number of functional tests that test high-level scenarios.

**Comment by artem-shelkovnikov** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.4|[](https://github.com/elastic/connectors-ruby/pull/262)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**